build.gradle.kts

```diff
@@ -0,0 +1,201 @@
+import org.ajoberstar.grgit.Grgit
+import com.github.spotbugs.*
+
+buildscript {
+    repositories {
+        jcenter()
+        maven {
+            setUrl("https://github.com/WPIRoboticsProjects/opencv-maven/raw/mvn-repo")
+        }
+    }
+    dependencies {
+        classpath(group = "de.dynamicfiles.projects.gradle.plugins", name = "javafx-gradle-plugin", version = "8.8.2")
+        classpath(group = "edu.wpi.first.wpilib.opencv", name = "opencv-installer", version = "2.0.0")
+    }
+}
+
+plugins {
+    `java`
+    `jacoco`
+    `checkstyle`
+    `pmd`
+    id("com.github.johnrengelman.shadow") version "4.0.3"
+    id("com.google.osdetector") version "1.4.0"
+    id("org.ajoberstar.grgit") version "2.0.0" apply false
+    id("net.ltgt.errorprone") version "0.0.16"
+    id("com.github.spotbugs") version "1.6.4"
+}
+
+repositories {
+    mavenCentral()
+    jcenter()
+}
+
+tasks.withType<Wrapper>().configureEach {
+    gradleVersion = "5.0"
+    distributionType = Wrapper.DistributionType.ALL
+}
+
+fun javaSubprojects(action: Project.() -> Unit) {
+    subprojects.minus(project(":ui:linuxLauncher")).forEach { project ->
+        project.action()
+    }
+}
+
+var grgit: Grgit? = null
+if (rootProject.file(".git").exists()) {
+    project.apply {
+        plugin("org.ajoberstar.grgit")
+    }
+    grgit = Grgit.open()
+}
+
+javaSubprojects {
+    apply {
+        plugin("java")
+        plugin("org.gradle.jacoco")
+        plugin("org.gradle.pmd")
+        plugin("org.gradle.checkstyle")
+        plugin("net.ltgt.errorprone")
+        plugin("com.github.spotbugs")
+    }
+    repositories {
+        mavenCentral()
+        jcenter()
+        maven {
+            name = "WPILib Maven Release"
+            setUrl("http://first.wpi.edu/FRC/roborio/maven/release")
```

Comment:
Booo. Not a fan of `http` here.

Can you use the new Gradle repository filers so that only artifacts that are actually on these servers get searched in these repos?

https://docs.gradle.org/current/userguide/declaring_repositories.html#declaring_a_repository_filter

Reply:
first.wpi.edu unfortunately does not have an HTTPS cert
The repository filter looks nice, though

Reply:
Darn, the filtering isn't available in 5.0

Reply:
Is there any sort of hash validation we can do before we just start arbitrarily running code that was downloaded insecurely?

Is there some way we can validate the SHA 256 hash of it first or something?

Reply:
I don't think there's anything we can do beyond what I assume Gradle does already with using the .md5 or .sha hash files in the Maven repo.

This is also out of scope for this PR

Reply:
Fair enough.